easypost/requestor.py

```diff
@@ -11,33 +13,41 @@
 
 
 class Requestor:
-    def __init__(self, local_api_key=None):
+    def __init__(self, local_api_key: str = None):
         self._api_key = local_api_key
 
     @classmethod
-    def _objects_to_ids(cls, param):
+    def _objects_to_ids(cls, param: dict) -> dict:
         if isinstance(param, EasyPostObject):
             return {"id": param.id}
         elif isinstance(param, dict):
-            return {k: cls._objects_to_ids(v) for k, v in param.items()}
-        elif isinstance(param, list):
-            return [cls._objects_to_ids(v) for v in param]
+            data = {}
+            for (k, v) in param.items():
+                if isinstance(v, list):
+                    data[k] = [cls._objects_to_ids(item) for item in v]  # type: ignore
+                else:
+                    data[k] = cls._objects_to_ids(v)  # type: ignore
+            return data
         else:
             return param
 
-    def request(self, method, url, params=None, api_key_required=True):
+    def request(
+        self, method: str, url: str, params: dict = None, api_key_required: bool = True
+    ) -> Tuple[dict, Optional[str]]:
```

Comment:
`Dict` (there are a ton of these, I'm going to stop explicitly marking them, please check this PR for any that need changing.